### ISSUE TITLE
Consider self-distributing the Firefox add-on

### ISSUE THREAD

**Issue by Rongronggg9** [DIYgod/RSSHub-Radar#691]
Context: https://github.com/DIYgod/RSSHub-Radar/pull/635#issuecomment-1100721069

#### TL;DR
> You really should read the context.

Add-ons distributed on AMO (https://addons.mozilla.org) are unable to use `eval()`, making remote rules updating and local rules modifications impossible. However, a self-distributed add-on faces much looser reviews so that's possible to enable both features.

<details>
<summary>You really should not read this</summary>

In fact, no reviews, but automatic checks and approval.

![](https://user-images.githubusercontent.com/15956627/163693953-3b38710f-1aab-49e3-ae03-29c1e9853a15.png)

Shhh...

</details>

#### An effective example
[rsshub_radar-1.7.1-fx.zip](https://github.com/DIYgod/RSSHub-Radar/files/8500722/rsshub_radar-1.7.1-fx.zip) (already signed by AMO so it can be installed easily, change the ext name to `xpi` if needed)
Modifications on local rules are no longer being ignored, making it possible to update rules by yourself, or debug your own rules before summiting a new RSSHub route, cheers!
As for remote rules, it is still disabled. But if you were a **magical** girl who loved a [**cyber chef**](https://gchq.github.io/CyberChef/), enabling it would be so easy.
```
P4xF1rYWBS+cSclbXFWDY4+ady5FUSK8PuHFYOM=bv7+p8J1poRKB9vcpW+cUWK1PzwySFwadNSlUx+1PugmpzPKavL+s+KAPr+mxrSJ5o5eovMApSPKYOvcB+n7XyxW
```

##### Disclaimer
- My self-distributed add-on comes with absolutely no warranty.
- So do remote rules. Technically speaking, enabling remote rules is allowing remote code execution (RCE), so **you are the only one responsible for your action**.
- The add-on contains some code from a drafting PR. I was just too lazy to remove it.
- You can audit the source code yourself: https://github.com/Rongronggg9/RSSHub-Radar/tree/ff

#### Related issues / PRs
Shouldn't #667 be enough? Probably not. Radar rules are now required for any new RSSHub routes. No reason to force a Firefox user to install Chromium or modify the code of RSSHub-Radar (then build and install on Firefox) before they can debug their radar rules, isn't it? What's more, a sideloaded add-on will be uninstalled automatically at quit (only Nightly, Beta, or ESR may escape if configured - but [<20%](https://en.wikipedia.org/wiki/Firefox_version_history) Firefox users use these three channels), so that's also an unacceptable solution for ordinary Firefox users.

Here's a contributor who had to waste a lot of time before realizing RSSHub-Radar is almost completely broken on Firefox (https://github.com/DIYgod/RSSHub/pull/9537).

Other related issues: #687 #660 #408 #460 #675

**Comment by Seele0oO** [comment]
Thank you for your contribution, which can now finally barely be used.

**Comment by simpread** [comment]
谜语人啊，第二段解密不出来QAQ

**Comment by Seele0oO** [comment]
https://github.com/Seele0oO/RSSHub-Radar
I did it.
Just click on the release xpi file to complete the installation.
I only make sure it works for regular users like me.
As well as it won't have live updates to upstream, I haven't finished writing the github action configuration yet. If anyone would like to submit a Pull Request appreciate it.

**Comment by DIYgod** [comment]
I am reconsidering self-distribution. A Firefox extension reviewer informed me that he couldn't reproduce the same results from our source code on his Ubuntu, leading him to remove the publication of our extensions. This is very ridiculous.

**Comment by enchained** [comment]
@DIYgod Is any of this related to why the Firefox version is now very outdated here https://addons.mozilla.org/firefox/addon/rsshub-radar/? I think I installed v2 from this store in the past (and web archive confirms it was available back in time), but now on another PC I can only install v1 from the Firefox store. This was very confusing, since no Firefox issues were mentioned in the Radar github readme. 

Is there a proper way I could permanently install an up-to-date Radar on regular Firefox (no Dev/Nightly)? Can the readme be edited to reflect the ongoing Firefox issues and their possible solutions?
